### PR TITLE
Send invites to WordPress.com sites

### PR DESCRIPTION
**Pull request by hypest**
Related to #3901 

Adds the ability to send invites to users to become members of the site.

To test:
* Go to "People" from the Site screen
* Tap the "+" button in the top-right
* Start typing usernames or emails in the top field
* Use "Enter" to input multiple usernames/emails. Focusing away from the input field should also "register" the username
* Usernames/email will turn red if there is any issue with them
* (Optional) Select a role
* (Optional) Enter a custom message to be send with the invite
* Tap "SEND" to send the invite. Send will stop if there are any invalid usernames/emails
* Screen will "clear" if SEND succeeds

Needs review: @bummytime

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/SlidingScrollView.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.wordpress.android.ui;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.ViewTreeObserver;
+import android.widget.ScrollView;
+
+public class SlidingScrollView extends ScrollView {
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/SlidingScrollView.java]: Might be worth pointing to `SlidingRelativeLayout` as a comment as they're using the same implementation.

Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/SlidingScrollView.java]: Not sure about pointing to `SlidingRelativeLayout` but, lemme add a small comment about the need for `SlidingScrollView` in the first place: So, the new class is needed in order to support proper scrolling up/down when the IME is open, when the user types in the `EditText`s. The whole view is rather tall and some is hidden when the IME opens. The `SlidingRelativeLayout `makes it harder to support the scrolling.

**Comment by bummytime** [comment]
Ok @hypest I tested this out on various API levels and devices. All in all, it appears to be working pretty well. I had two UX comments:

1) I see if an error occurs sending an invite out, the failed names are left behind in the usernames field:

napkin 31 06-03-16 12 40 18 pm

Should we call this out to the user...or do you think it is pretty obvious?

2) If the user enters their own name (like in the image above) should we validate that out before they even hit the send button?

Looking over the code next....

**Comment by bummytime** [comment]
Code looks pretty good @hypest. Let me know your thoughts on the UX-y stuff above. Not deal breakers, but they did jump out at me. 😄

**Comment by hypest** [comment]
Thanks for the review @bummytime !

> I see if an error occurs sending an invite out, the failed names are left behind in the usernames field:
> Should we call this out to the user...or do you think it is pretty obvious?

I'd say that it is not really obvious that it's the errored usernames that were left behind, although there is a hint in the Toast (`...some recipients`). I will try to make it more clear.

> If the user enters their own name (like in the image above) should we validate that out before they even hit the send button?

Not sure what you mean here. If user hits `space` or `enter` at the end of the username, the string will be buttonised and validated. We can't be sure that the user has finished typing in the username before tapping space/enter or the SEND button. So, we can't validate it yet.

If user ends up hitting "SEND" without a space/enter before, the app actually buttonises the username, triggers the validation, waits for the remote calls to finish and will not do any sending if an invalid username was entered (which includes user's own username). 

Perhaps you mean to prevent the own username to be inputted, like we do with duplicate usernames (where the username inputted just vanishes when user taps space/enter)? Lemme know what you have in mind.

**Comment by hypest** [comment]
@bummytime I've addressed your `...failed names are left behind...` point by keeping the Toast message minimal (though I hope it does convey the message that the invite was indeed sent anyway) and adding error message lines for the usernames that failed. I think this provides enough context for the user to understand why some usernames were left behind and perhaps try again or not.

The PR is ready for another pass at this point. Thanks!

**Comment by bummytime** [comment]
@hypest 

> Perhaps you mean to prevent the own username to be inputted, like we do with duplicate usernames (where the username inputted just vanishes when user taps space/enter)? Lemme know what you have in mind.

That is exactly what I meant and said poorly :-)

**Comment by hypest** [comment]
@bummytime actually, I think I'm going to leave it as it is and not prevent from inputting the own username. One reason is that I don't think it will be as obvious why it is not allowed and would need a Toast to explain it in brief. One could almost say the same for the duplicate username clearing but in that case the first occurrence of the username is there and the user can review his inputs.

Second reason is that Calypso does allow the own username to be inputted and it errors it out by a "Already a member" message, and that's exactly what the app does atm. Parity with Calypso is quite a driving force so, I'd say let's stick with the current behaviour for the time being.

Lemme know if there is some other detail that needs attention in the PR, thanks!

**Comment by bummytime** [comment]
Sounds good to me @hypest! :shipit:  Thanks!

**Pull request merged**